tools/convertApacheAccessLogFile.php

```diff
@@ -0,0 +1,1097 @@
+<?php
+
+/**
+ * @file tools/convertApacheAccessLogFile.php
+ *
+ * Copyright (c) 2022 Simon Fraser University
+ * Copyright (c) 2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ConvertApacheAccessLogFile
+ * @ingroup tools
+ *
+ * @brief CLI tool to copy, prepare and convert apache access log file into the new format needed for stats reprocessing.
+ *
+ * The file will be copied to the {files_dir}/usageStats/tmp/ folder,
+ * only entries related to the current instalation will be filtered,
+ * the file will be spit by day,
+ * renamed into apache_usage_events_YYYYMMDD.log,
+ * converted to the new format,
+ * and copied into the {files_dir}/usageStats/archive/ folder.
+ *
+ * Special cases from the release 2.x are handled as following:
+ *
+ * Issue Galley:
+ * with PDF viewer:
+ * issue/viewIssue/issueId/galleyId followed by issue/viewFile/issueId/galleyId
+ * -> only issue/viewFile/issueId/galleyId will be considered.
+ * There is also only issue/download/issueId/galleyId (when download link is used).
+ * without PDF viewer:
+ * issue/viewIssue/issueId/galleyId will not be considered because the file is actually not downloaded.
+ * But issue/download/issueId/galleyId will be considered.
+ *
+ * PDF Galley:
+ * article/view/articleId/galleyId followed by article/viewFile/articleId/galleyId
+ * -> only article/viewFile/articleId/galleyId will be considered.
+ * There is also only article/donwload/articleId/galleyId (when download link is used.
+ * without PDF viewer:
+ * article/view/articleId/galleyId will not be considered because the file is actually not downloaded.
+ * But article/download/articleId/galleyId will be considered.
+ *
+ * HMTL Galley:
+ * article/view/articleId/galleyId followed by article/viewFile/articleId/galleyId
+ * -> only article/viewFile/articleId/galleyId will be considered.
+ *
+ * Other and Remote Galley:
+ * article/view/articleId/galleyId
+ *
+ * Supp File:
+ * article/downloadSuppFile/articleId/galleyId
+ */
+
+require(dirname(__FILE__, 4) . '/tools/bootstrap.php');
+
+use APP\core\Application;
+use APP\facades\Repo;
+use APP\statistics\StatisticsHelper;
+use PKP\cliTool\ConvertLogFile;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\file\FileManager;
+use PKP\statistics\PKPStatisticsHelper;
+use PKP\submission\Genre;
+use PKP\task\FileLoader;
+
+class ConvertApacheAccessLogFile extends ConvertLogFile
+{
+    /**
+     * Path to the egrep program, required for this tool to work, e.g. '/bin/egrep'
+     */
+    public const EGREP_PATH = '';
```

Comment:
Did I miss where this is defined e.g. in a subclass?

Reply:
Ah, I wanted to force the user to enter it here but it is better to have a default value here...